backend/integraflow/graphql/user/mutations/authentication/google_user_auth_challenge.py

```diff
@@ -0,0 +1,105 @@
+from typing import Dict
+
+import graphene
+from django.conf import settings
+from django.core.exceptions import ValidationError
+from google.auth.transport import requests
+from google.oauth2 import id_token
+from google_auth_oauthlib.flow import Flow
+
+from integraflow.core.jwt import create_access_token, create_refresh_token
+from integraflow.graphql.core import ResolveInfo
+from integraflow.graphql.core.doc_category import DOC_CATEGORY_AUTH
+from integraflow.graphql.core.mutations import BaseMutation
+from integraflow.graphql.core.types.common import UserError
+from integraflow.user.models import User
+
+GOOGLE_AUTH_CLIENT_CREDENTIALS = settings.GOOGLE_AUTH_CLIENT_CREDENTIALS
+
+
+class GoogleUserAuthChallenge(BaseMutation):
```

Comment:
@AfeezGL Can you rename this mutation to `GoogleUserAuth`. The `challenge` part doesn't tell the whole story because it has been taken care of by the frontend.